broadinstitute/cromwell
# WDL 1.0 upgrade script Phase I: WDL writer

Takes WDL 1.0 case classes (`wdl.model.draft3.elements.FileElement` and its children) and writes them out as WDL source code.

Test scheme:
1. Load the WDL as case classes
2. Write them out as source
3. Load up the source from step 2 as case classes
4. Verify the case classes from (1) and (3) are identical

I also uncovered some gaps in the set of WDLs we use for unit tests - I had gotten all the tests to pass, but there were still a handful of `???`'s in my code. Added test cases such that all of them were hit (and then implemented).
